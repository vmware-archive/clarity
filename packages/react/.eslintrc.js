const OFF = 0;
const WARN = 1;
const ERROR = 2;

const bannedTSTypes = {
  Array: 'Use [] instead.',
  Object: 'Use {} instead.',
  Boolean: 'Use `boolean` instead.',
  Number: 'Use `number` instead.',
  String: 'Use `string` instead.',
};

// only adding rules that override the defaults or enforce new standards
const rules = {
  '@typescript-eslint/explicit-function-return-type': OFF, // too much work at the moment
  '@typescript-eslint/no-explicit-any': OFF, // we have custom events whose type is any; revisit if needed
  '@typescript-eslint/no-use-before-define': OFF,
  '@typescript-eslint/ban-types': [ERROR, { types: bannedTSTypes }],
  '@typescript-eslint/explicit-member-accessibility': [ERROR, { accessibility: 'no-public' }],
  '@typescript-eslint/no-unused-vars': [
    ERROR,
    {
      varsIgnorePattern: 'React|Cds',
    },
  ],
  'import/extensions': [ERROR, 'ignorePackages'],
};

const parserOptions = {
  project: 'tsconfig.eslint.json',
  tsconfigRootDir: __dirname,
  sourceType: 'module',
};

const overrides = [
  {
    files: ['*.test.tsx'],
    rules: {
      'import/extensions': OFF,
    },
  },
];

const config = {
  extends: '../../.eslintrc.js',
  parserOptions,
  rules,
  overrides,
};

module.exports = config;
