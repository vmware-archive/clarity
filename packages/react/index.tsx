import React from 'react';
import ReactDOM from 'react-dom';
import 'normalize.css/normalize.css';
import '@cds/core/global.min.css';
import '@cds/city/css/bundles/default.min.css';
import App from './App';

ReactDOM.render(<App />, document.getElementById('root'));
