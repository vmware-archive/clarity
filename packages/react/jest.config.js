module.exports = {
  roots: ['./'],
  globals: {
    'ts-jest': {
      tsConfig: 'tsconfig.test.json',
    },
  },
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  moduleNameMapper: {
    '@cds/(.*)': '<rootDir>/../core/dist/$1',
  },
  setupFilesAfterEnv: ['./jest.setup.ts'],
  snapshotSerializers: ['enzyme-to-json/serializer'],
  testMatch: ['**/__tests__/**/*.+(ts|tsx|js)', '**/?(*.)+(test).+(ts|tsx|js)'],
  transform: {
    '^.+\\.(js|jsx)$': 'babel-jest',
    '^.+\\.(ts|tsx)$': 'ts-jest',
  },
  transformIgnorePatterns: ['node_modules/(?!@cds/core|lit-element|lit-html|ramda)'],
};
