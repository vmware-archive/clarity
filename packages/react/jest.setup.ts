import '@testing-library/jest-dom/extend-expect';

if (typeof window !== 'undefined') {
  // jsdom doesn't implment IntersectionObserver
  // https://github.com/jsdom/jsdom/issues/2032
  const mockObserverAPI = () =>
    jest.fn().mockReturnValue({
      observe: jest.fn(),
      unobserve: jest.fn(),
      disconnect: jest.fn(),
    });

  window.ResizeObserver = mockObserverAPI();
  window.IntersectionObserver = mockObserverAPI();

  // jsdom implements getBoundingClientRect, but not DOMRect
  if (!window.DOMRect) {
    const domRectMock = () => jest.fn().mockReturnValue({});
    window.DOMRect = (domRectMock() as unknown) as typeof window.DOMRect;
  }
}
