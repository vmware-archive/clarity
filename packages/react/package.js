'use strict';

const fs = require('fs-extra');
const path = require('path');
const del = require('del');

// Temporary script needed to remove cache files and
// files Angular CLI leaves in referenced dependencies
// https://github.com/ng-packagr/ng-packagr/pull/1372
// https://github.com/ng-packagr/ng-packagr/issues/1318
// https://github.com/angular/angular/issues/33395
// https://justinfagnani.com/2019/11/01/how-to-publish-web-components-to-npm/
const read = dir =>
  fs
    .readdirSync(dir)
    .reduce(
      (files, file) =>
        fs.statSync(path.join(dir, file)).isDirectory()
          ? files.concat(read(path.join(dir, file)))
          : files.concat(path.join(dir, file)),
      []
    );

del.sync(
  [
    './dist/react/**/*.{tsbuildinfo,test.js,test.js.map,test.d.ts,.snap}',
    './dist/react/*.{tsbuildinfo,test.js,test.js.map,test.d.ts,.snap}',
  ],
  { force: true }
);

read('./dist/react')
  .filter(f => f.includes('package.json'))
  .forEach(file => {
    const packageFile = fs.readJsonSync(file);
    const metaData = {
      main: './index.js',
      module: './index.js',
      typings: './index.d.ts',
      type: 'module',
    };

    if (file === 'dist/react/package.json') {
      ['alias', 'browserslist', 'scripts', 'devDependencies'].forEach(p => delete packageFile[p]);
      // move @clr/core from dependencies to peerDependencies and assign version number that matches the package.json
      packageFile.peerDependencies = { '@clr/core': `^${packageFile.version}` };
      delete packageFile.dependencies['@clr/core'];
    }

    fs.writeJsonSync(file, { ...packageFile, ...metaData }, { spaces: 2 });
  });
