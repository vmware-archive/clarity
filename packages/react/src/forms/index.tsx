import { CdsControlMessage as ControlMessage } from '@clr/core/forms';
import { CdsControlAction as ControlAction } from '@clr/core/forms';
import { CdsInternalControlGroup as ControlGroup } from '@clr/core/forms';
import { CdsControl as Control } from '@clr/core/forms';
import { CdsFormGroup as FormGroup } from '@clr/core/forms';

import '@clr/core/forms/register';
import { createReactComponent } from '../converter/react-wrapper';

type CdsControlMessageType = ControlMessage;
type CdsControlActionType = ControlAction;
type CdsControlGroupType = ControlGroup;
type CdsControlType = Control;
type CdsFormGroupType = FormGroup;

export class CdsControlMessage extends createReactComponent<CdsControlMessageType>('cds-control-message') {}
export class CdsControlAction extends createReactComponent<CdsControlActionType>('cds-control-action') {}
export class CdsControlGroup extends createReactComponent<CdsControlGroupType>('cds-control-group') {}
export class CdsControl extends createReactComponent<CdsControlType>('cds-control') {}
export class CdsFormGroup extends createReactComponent<CdsFormGroupType>('cds-form-group') {}
