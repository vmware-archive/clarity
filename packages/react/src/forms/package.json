{
    "sideEffects": true,
    "name": "@cds/react/forms"
  }
