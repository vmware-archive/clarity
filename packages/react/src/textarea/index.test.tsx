import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { CdsTextarea } from './index';
import { CdsFormGroup, CdsControlMessage } from '../forms/index';

describe('CdsTextarea', () => {
  it('renders', () => {
    const wrapper = shallow(
      <div>
        <CdsFormGroup layout="vertical">
          <CdsTextarea layout="vertical">
            <label>label</label>
            <textarea></textarea>
            <CdsControlMessage>message text</CdsControlMessage>
          </CdsTextarea>
        </CdsFormGroup>
      </div>
    );

    const renderedComponent = wrapper.find('CdsTextarea');
    expect(renderedComponent.at(0).html()).toMatch(/label/);
    expect(renderedComponent.at(0).html()).toMatch(/message text/);
  });

  it('snapshot', () => {
    const wrapper = mount(
      <div>
        <h3>Vertical</h3>
        <CdsFormGroup layout="vertical">
          <CdsTextarea layout="vertical">
            <label>label</label>
            <textarea></textarea>
            <CdsControlMessage>message text</CdsControlMessage>
          </CdsTextarea>

          <CdsTextarea layout="vertical" status="error">
            <label>error status</label>
            <textarea></textarea>
            <CdsControlMessage status="error">error message</CdsControlMessage>
          </CdsTextarea>

          <CdsTextarea layout="vertical" status="success">
            <label>success status</label>
            <textarea></textarea>
            <CdsControlMessage status="success">success message</CdsControlMessage>
          </CdsTextarea>
        </CdsFormGroup>

        <h3>Horizontal</h3>
        <CdsFormGroup layout="horizontal">
          <CdsTextarea layout="horizontal">
            <label>label</label>
            <textarea></textarea>
            <CdsControlMessage>message text</CdsControlMessage>
          </CdsTextarea>

          <CdsTextarea layout="horizontal" status="error">
            <label>error status</label>
            <textarea></textarea>
            <CdsControlMessage status="error">error message</CdsControlMessage>
          </CdsTextarea>

          <CdsTextarea layout="horizontal" status="success">
            <label>success status</label>
            <textarea></textarea>
            <CdsControlMessage status="success">success message</CdsControlMessage>
          </CdsTextarea>
        </CdsFormGroup>
      </div>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
