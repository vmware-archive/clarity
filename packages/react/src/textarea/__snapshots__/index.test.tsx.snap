// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdsTextarea snapshot 1`] = `
<div>
  <h3>
    Vertical
  </h3>
  <CdsFormGroup
    layout="vertical"
  >
    <cds-form-group>
      <CdsTextarea
        layout="vertical"
      >
        <cds-textarea>
          <label>
            label
          </label>
          <textarea />
          <CdsControlMessage>
            <cds-control-message>
              message text
            </cds-control-message>
          </CdsControlMessage>
        </cds-textarea>
      </CdsTextarea>
      <CdsTextarea
        layout="vertical"
        status="error"
      >
        <cds-textarea>
          <label>
            error status
          </label>
          <textarea />
          <CdsControlMessage
            status="error"
          >
            <cds-control-message>
              error message
            </cds-control-message>
          </CdsControlMessage>
        </cds-textarea>
      </CdsTextarea>
      <CdsTextarea
        layout="vertical"
        status="success"
      >
        <cds-textarea>
          <label>
            success status
          </label>
          <textarea />
          <CdsControlMessage
            status="success"
          >
            <cds-control-message>
              success message
            </cds-control-message>
          </CdsControlMessage>
        </cds-textarea>
      </CdsTextarea>
    </cds-form-group>
  </CdsFormGroup>
  <h3>
    Horizontal
  </h3>
  <CdsFormGroup
    layout="horizontal"
  >
    <cds-form-group>
      <CdsTextarea
        layout="horizontal"
      >
        <cds-textarea>
          <label>
            label
          </label>
          <textarea />
          <CdsControlMessage>
            <cds-control-message>
              message text
            </cds-control-message>
          </CdsControlMessage>
        </cds-textarea>
      </CdsTextarea>
      <CdsTextarea
        layout="horizontal"
        status="error"
      >
        <cds-textarea>
          <label>
            error status
          </label>
          <textarea />
          <CdsControlMessage
            status="error"
          >
            <cds-control-message>
              error message
            </cds-control-message>
          </CdsControlMessage>
        </cds-textarea>
      </CdsTextarea>
      <CdsTextarea
        layout="horizontal"
        status="success"
      >
        <cds-textarea>
          <label>
            success status
          </label>
          <textarea />
          <CdsControlMessage
            status="success"
          >
            <cds-control-message>
              success message
            </cds-control-message>
          </CdsControlMessage>
        </cds-textarea>
      </CdsTextarea>
    </cds-form-group>
  </CdsFormGroup>
</div>
`;
